versions/1.21.5/src/main/java/at/hannibal2/skyhanni/utils/compat/OrdereredTextUtils.kt

```diff
@@ -93,24 +93,31 @@ object OrderedTextUtils {
 
         return OrderedText { visitor ->
             var lastStyle = Style.EMPTY
-            var wasLastStyle = false
-
-            for (char in legacyString ?: "") {
-
-                if (char == '§') {
-                    wasLastStyle = true
-                } else if (wasLastStyle) {
-
-                    val formatting = Formatting.byCode(char)
+            val str = legacyString ?: ""
+            var i = 0
+            while (i < str.length) {
+                val c = str[i]
+                if (c == '§' && i + 1 < str.length) {
+                    val code = str[i + 1]
+                    if (code == '#' && i + 7 < str.length) {
+                        val hexPart = str.substring(i + 2, i + 8)
+                        val rgb = hexPart.toIntOrNull(16)
+                        if (rgb != null) {
+                            lastStyle = lastStyle.withColor(TextColor.fromRgb(rgb))
```

Comment:
probably better to do this
```suggestion
                            lastStyle = lastStyle.withColor(rgb)
```